manuscript/10-simple-rules-for-teaching-Data-Science.qmd

```diff
@@ -23,209 +23,218 @@ fig-cap-location: bottom
-In her cheat sheet, she teaches all the many different joins from the `{dplyr}` [@dplyr] package.
-This is a difficult topic for students to understand and memorize, and when instructors teach this with large data sets it is really hard for students to get an idea of what's going on and all these joins that are also similarly named (e.g, left join, right join, inner join, outer join, etc).
-To address this issue, Jenny's cheat sheet uses two toy data sets on super hero comic characters and publishers.
-The super hero data set has only 7 rows and 4 columns, while the publisher data set has 3 rows and 2 columns.
-These data sets are small enough The cheat sheet thenb goes through all the different possible joins and narrates and shows the output of each.
-For learners this becomes much more tractable and easy to understand.
+Our fourth simple rule is to use tractable or toy data examples when introducing a new tool, method, or algorithm to students.
+Tractable or toy data sets have a countable number of elements, things that can fit inside our working memory.
+This allows students to track the progression of everything through the different steps of the algorithm, to see how the elements are manipulated, and to gain a deeper understanding of these concepts.
+For example, in one of our courses, we use the Palmer penguins data set [@horst2020palmerpenguins] to introduce the students to k-means clustering.
+Instead of giving them the entire data set, which contains hundreds of observations, we first subset the data to just a handful of observations.
+Then we can walk the students through what happens to these observations at each step of the algorithm, building an understanding and intuition for the algorithm.
+Inspiration from this comes from Jenny Bryan's great dplyr joins cheat sheet [@bryan2015stat545].
+In her cheat sheet, she teaches all the various joins from the dplyr package [@dplyr].
+This is a difficult topic for students to understand and memorize, and when instructors teach this with large data sets, it is really hard for students to get an idea of what's going on and all these joins that are also similarly named (e.g, left join, right join, inner join, outer join, etc).
+To address this issue, Jenny's cheat sheet uses two toy data sets on superhero comic characters and publishers.
+The superhero dataset has only 7 rows and 4 columns, while the publisher dataset has 3 rows and 2 columns.
+These data sets are small enough. The cheat sheet then outlines all the possible joins, narrating and displaying the output of each.
+For learners, this becomes much more tractable and easy to understand.
+
+It is, of course, not feasible to keep learners interested and motivated to learn more if we only use toy data sets.
+So, once students have a conceptual understanding of the tool, method, or algorithm being taught, we can then move on to using real and rich data sets, which is our next rule.
```

Comment:
Great bridge here to the next rule! And maybe this is already enough to address to contradiction between these two rules that you pointed out?